public/js/simpleStrings-0.0.1.js

```diff
@@ -31,7 +31,7 @@
 
       function translate(item) {
         item = $(item);
-        var text = item.text();
+        var text = item.clone().children().remove().end().text();
```

Comment:
Ah! I think I've finally figured out why you're doing this. Given this example,

```svg
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="60" height="20">
  <g transform="translate(10,20)">
    <text><tspan style="fill:#ff0000">{{red}}</tspan><tspan style="fill:#00ff00">{{green}}</tspan></text>
  </g>
</svg>
```

The original code would have matched the outermost `<text>` tag and evaluated the text value it and all children, or `{{red}}{{green}}`. This code means that that match would have evaluated *nothing*, but then the `<tspan>` tags would then each be individually matched and then evaluated.

Is that a correct assessment of the intent? If so, I'll totally merge this. This is a better solution than my original which required you to make separate text elements.

Reply:
Actually, no. Am going to have to think about this more. This solution will work perfectly well when there’s no actual text in the text element, but as soon as there’s a replacement token that’s outside the `tspan` element(s), then the call to `text(replacement)` will replace all children🙃.

Lemme think on this a bit more. If we required that the token filled the entire element, we’d probably be ok, but I can see why you want to enable multiple tokens in a string.

Reply:
I'm going to rewrite this whole routine using `.contents()` to get the text nodes as well as any children. Then I'll just translate them individually. Thanks for running down this rabbit hole with me., I think we'll come out of this with a solid solution.

Reply:
ok, I'm fine with that :wink: